src/server/game/Spells/SpellEffects.cpp

```diff
@@ -4708,9 +4708,10 @@ void Spell::EffectKnockBack(SpellEffIndex effIndex)
     if (!unitTarget)
         return;
 
-    if (Creature* creatureTarget = unitTarget->ToCreature())
-        if (creatureTarget->isWorldBoss() || creatureTarget->IsDungeonBoss())
-            return;
+    if (m_caster->GetTypeId() == TYPEID_PLAYER || m_caster->GetOwnerGUID().IsPlayer() || m_caster->IsHunterPet())
```

Comment:
@Shauren thx, really
i looked more closely and found that kicking mechanics -- MECHANIC_INTERRUPT
eg spells 1766, 32846 and so on

so shouldbe 
(info->MechanicImmuneMask & __MECHANIC_INTERRUPT__)
instead of 
(info->MechanicImmuneMask & MECHANIC_KNOCKOUT)